src/commands/gen-js/library.ts

```diff
@@ -61,15 +61,18 @@ export function genJS(
       let trackCall = ''
       let validateCall = ''
       if (client === Client.js) {
-        parameters = 'props = {}, context'
-        trackCall = `this.analytics.track('${name}', props, genOptions(context))`
+        parameters = 'props = {}, options'
+        trackCall = `this.analytics.track('${name}', props, {
+          ...options,
+          context: this.addTypewriterContext(options.context)
```

Comment:
Ah, so this will allow you to use it as a top-level object _or_ as a `context` key in `options`. 👍